File: backend-gin/internal/lib/types_/pagination.go
package types_

type PaginatedData[T any] struct {
	Page       int `json:"page" example:"1"`
	TotalPages int `json:"total_pages" example:"2"`
	TotalCount int `json:"total_count" example:"40"`
	Data       []T `json:"data"`
}

type PaginatedDict = PaginatedData[map[string]any]

type PaginationData struct {
	Page int
	Size int
}

func (p *PaginationData) Skip() int {
	return (p.Page - 1) * p.Size
}

File: backend-axum/src/lib/types_/pagination.rs
use serde::Serialize;
use std::fmt::Debug;
use utoipa::ToSchema;

#[derive(Debug, Serialize, ToSchema)]
#[serde(rename_all = "camelCase")]
pub struct PaginatedData<T: Debug + Serialize + ToSchema> {
    #[schema(example = "2")]
    pub page: usize,
    #[schema(example = "10")]
    pub total_pages: usize,
    #[schema(example = "197")]
    pub total_count: usize,
    pub data: Vec<T>,
}
